sample/src/main/java/com/microsoft/gctoolkit/sample/collections/XYDataSet.java

```diff
@@ -49,8 +49,15 @@ public XYDataSet scaleSeries(double scaleFactor) {
         return scaled;
     }
 
-    public double max() {
-        return dataSeries.stream().map(Point::getY).mapToDouble(Number::doubleValue).max().getAsDouble();
+    /**
+     * Returns the largest Y value in the XYDataSet as an OptionalDouble,
+     * with an empty optional if the dataset is empty.
+     */
+    public OptionalDouble maxOfY() {
```

Comment:
curious to know why the move to Optional?

Reply:
The previous code would throw an exception if you run it with an empty XYDataSet. In other words, new XYDataSet().max() throws an exception.